lib/Listeners/LoadPublicViewerListener.php

```diff
@@ -27,22 +27,55 @@
 namespace OCA\Files_PDFViewer\Listeners;
 
 use OCA\Files_PDFViewer\AppInfo\Application;
-use OCP\AppFramework\Http\Events\BeforeTemplateRenderedEvent;
+use OCA\Files_Sharing\Event\BeforeTemplateRenderedEvent;
 use OCP\AppFramework\Http\TemplateResponse;
+use OCP\AppFramework\QueryException;
 use OCP\EventDispatcher\Event;
 use OCP\EventDispatcher\IEventListener;
+use OCP\IServerContainer;
 use OCP\Util;
 
 class LoadPublicViewerListener implements IEventListener {
+
+	/** @var IServerContainer */
+	private $serverContainer;
+
+	public function __construct(IServerContainer $serverContainer) {
+		$this->serverContainer = $serverContainer;
+	}
+
 	public function handle(Event $event): void {
 		if (!$event instanceof BeforeTemplateRenderedEvent) {
 			return;
 		}
 
-		// Make sure we are on a public page rendering
-		if ($event->getResponse()->getRenderAs() !== TemplateResponse::RENDER_AS_PUBLIC) {
+		// If the event has a scope it is not the default share page but, for
+		// example, the authentication page
+		if ($event->getScope() !== null) {
+			return;
```

Comment:
Are we sure no other area uses the Template without a scope?

Reply:
In server `OCA\Files_Sharing\Event\BeforeTemplateRenderedEvent` is emitted only by the [ShareController](https://github.com/nextcloud/server/blob/9d26671f05275055619960a66981201ed660bdd2/apps/files_sharing/lib/Controller/ShareController.php).

Is it emitted by any other app? I do not know, but in that case it would be the same as before, the JavaScript code would check if it is a public share page and if it is not the PDF viewer will not be loaded.